array_api_tests/test_linalg.py

```diff
@@ -925,11 +938,8 @@ def _test_vecdot(namespace, x1, x2, data):
     ph.assert_result_shape("vecdot", in_shapes=[x1.shape, x2.shape],
                            out_shape=res.shape, expected=expected_shape)
 
-    if x1.dtype in dh.int_dtypes:
-        def true_val(x, y, axis=-1):
-            return xp.sum(xp.multiply(x, y), dtype=res.dtype)
-    else:
-        true_val = None
+    def true_val(x, y, axis=-1):
+        return xp.sum(xp.multiply(_conj(x), y), dtype=res.dtype)
```

Comment:
Wait, what: https://github.com/data-apis/array-api-tests/blob/master/array_api_tests/test_linalg.py#L109 and I was sure this PR switches on equality testing. But `assert_equal` does not do what the implies:
https://github.com/data-apis/array-api-tests/blob/master/array_api_tests/test_linalg.py#L49

Reply:
Yes, I originally had it there but later had to comment it out when I found out that you fundamentally can't compare floating point stacks for some functions like eigh because sometimes implementations (like CuPy) use different algorithms that give different (but still mathematically correct) results. https://github.com/data-apis/array-api-tests/pull/101#issuecomment-1111457823 I never renamed the functions. While I did turn the floating-point checks off completely, and they don't make sense for some functions like eigh, they do make sense for functions like vecdot and others where the mathematical answer is has a single well-defined value. We just need to be careful about loss of significance, especially for the tensor contraction functions (vecdot, matmul, tensordot, etc.) that involve additions.